### PR TITLE
feat(demos): update to webpack 5

### PR DESCRIPTION
**Pull request by unstubbable**

### CODE REVIEW

File: packages/demos/package.json [hunk 1]
```diff
@@ -3,6 +3,13 @@
   "version": "2.8.1",
   "private": true,
   "license": "MIT",
+  "workspaces": {
+    "nohoist": [
+      "@types/webpack-dev-middleware",
+      "webpack",
+      "ts-loader"
+    ]
+  },
```
Comment (unstubbable) [inline packages/demos/package.json:12]: Our augmented types for `webpack` (`packages/demos/types/webpack.d.ts`) are not respected by TypeScript, unless we prevent the two packages that need them (as well as webpack itself) from being hoisted to the root.

File: .nvmrc [hunk 2]
```diff
@@ -1 +1 @@
-10
+12
```
Comment (clebert) [inline .nvmrc:1]: I think we can remove Node.js support for v10 (2018) from CI as well. Maintenance end of that version is in a little more than a month: 2021-04-30

File: packages/demos/src/webpack-base-config.js [hunk 3]
```diff
@@ -43,6 +43,7 @@ const webpackBaseConfig = {
   },
   plugins: [
     new webpack.DefinePlugin({
+      'process.env.BLUEPRINT_NAMESPACE': '""',
```
Comment (clebert) [inline packages/demos/src/webpack-base-config.js:46]: Why is the ENV var new?

File: .nvmrc (hunk 2 above)
Comment (unstubbable) [reply .nvmrc:1]: Let's do that in another PR.

File: packages/demos/src/webpack-base-config.js (hunk 3 above)
Comment (unstubbable) [reply packages/demos/src/webpack-base-config.js:46]: [Webpack 5 does no longer include a polyfill for `process`](https://webpack.js.org/migrate/5/#run-a-single-build-and-follow-advises). This env variable was/is accessed by blueprint (and they fall back to a default value if it is falsy).

File: .nvmrc (hunk 2 above)
Comment (clebert) [reply .nvmrc:1]: See, #648

**Review by clebert — APPROVED** [review]

**Pull request merged**